### PR TITLE
Create manual deploy for docker images on failure

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

Modify the CI to be able to manually run the docker image deploy job on failure of a job in the previous stages of the pipeline.

Here is an example of a pipeline with this thing activated, with fake job to show what happens when we have a failure.
https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/20841976

### Motivation

To not have to painfully create pipeline on `public-images` to deploy docker images when one job that should not prevent from releasing the image fail

### Additional Notes

The solution I propose is kinda hacky, I did not find a way to allow the existing job to be manually triggered when a previous job failed. As a workaround I created new manual jobs that depends on a job that will run only `on_failure`. If the pipeline succeed we should not be able to run manually the deployment of docker images

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -590,6 +620,19 @@ workflow:
       DSD_REPOSITORY: ci/datadog-agent/dogstatsd-release
       IMG_REGISTRIES: internal-aws-ddbuild
 
+.on_deploy_a7_internal_rc_manual:
+  - <<: *if_not_version_7
+    when: never
+  - <<: *if_not_deploy
+    when: never
+  - <<: *if_rc_tag_on_beta_repo_branch
+    when: on_success
```
Comment (chouetz) [inline .gitlab-ci.yml:629]:
```suggestion
    when: manual
```
I suppose it's exactly the same as `on_deploy_a7_rc_manual` but the variables?

File: .gitlab/deploy_containers.yml [hunk 2]
```diff
@@ -180,3 +180,67 @@ deploy_containers_latest-dogstatsd:
...
+    VERSION: 7-rc
+  needs: ["check_failure_for_manual"]
+
+
+deploy_containers-a7-rc_on_failure:
+  extends: .deploy_containers-a7_external
+  rules:
+    !reference [.on_deploy_a7_rc_manual]
+  variables:
+    VERSION: 7-rc
```
Comment (chouetz) [inline .gitlab/deploy_containers.yml:209]:
```suggestion
    VERSION: 7-rc
  needs: ["check_failure_for_manual"]
```
I assume it's the same for all manual jobs, they need the `check_failure_for_manual` as prerequisite?

File: .gitlab/deploy_containers.yml [hunk 3]
```diff
@@ -180,3 +180,67 @@ deploy_containers_latest-dogstatsd:
...
+
+
+deploy_containers-a6_on_failure:
+  extends: .deploy_containers-a6-base
+  rules:
+    !reference [.on_deploy_a6_rc_manual]
+  needs: ["check_failure_for_manual"]
+
+
+deploy_containers-a6-rc_on_failure:
```
Comment (chouetz) [inline .gitlab/deploy_containers.yml:239]: ❓ 
I guess there is no need for an internal deploy for agent6?

**Review by chouetz — APPROVED** [review]
I let you tell me if my comments are appropriate or not

File: .gitlab-ci.yml (hunk 1 above)
Comment (KevinFairise2) [reply .gitlab-ci.yml:629]: Yes oops

File: .gitlab/deploy_containers.yml (hunk 3 above)
Comment (KevinFairise2) [reply .gitlab/deploy_containers.yml:240]: Actually this job does not exist so I did not create it the manual one in case of failure

**Pull request closed without merging**